### PR TITLE
Added new feature Schur_Number

### PR DESCRIPTION
**Pull request by slacker404**
The Schur number S(k) is the largest integer n for which the interval  [1,n] can be partitioned into k sum-free sets.  http://mathworld.wolfram.com/SchurNumber.html

I have also made the partition which can be proven by induction and I have added test cases

### CODE REVIEW

File: sympy/combinatorics/tests/test_schur_number.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+"""
+Checks if every subset is sum-free(There are no x,y,z in the subset such that
+x + y = z)
+"""
+def _sum_free_test(subset):
+    for i in subset:
+        for j in subset:
+            if i + j in subset:
+                raise AssertionError("This subset is not sum free")
+    return "Everything is correct"
```
Comment (asmeurer) [inline sympy/combinatorics/tests/test_schur_number.py]: This line is not necessary

File: sympy/combinatorics/tests/test_schur_number.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+n = random.randint(1,1000)
+result = schur_partition(n)
+
+"""
+Checks if the occurance of all numbers  is exactly one
+"""
+t = 0
```
Comment (asmeurer) [inline sympy/combinatorics/tests/test_schur_number.py]: Tests should go inside of a test function. See how the other test files in SymPy are organized.

**Comment by asmeurer** [comment]
Would it also be useful for this function to work symbolically?

**Comment by slacker404** [comment]
I have changed the files now and I added the symbolic representation of the inequality. I think that the result of the function cannot be represented symbolically because  there is no formula derived except from the last subset which has numbers of the form 3*n + 1

**Comment by czgdp1807** [comment]
It looks like a nice feature. @slacker404 Would you like to continue your work on this?

**Comment by sachin-4099** [comment]
I would like to work on this. Please guide as to what is remaining to be implemented.
@asmeurer

**Comment by mijo2** [comment]
I would like to revive this PR, and I am planning to do it today. @sachin-4099 We can work on this together if its ok with you, it is fine by me.

**Comment by sachin-4099** [comment]
> I would like to revive this PR, and I am planning to do it today. @sachin-4099 We can work on this together if its ok with you, it is fine by me.

It is fine by me..

**Comment by czgdp1807** [comment]
Closing in favour of https://github.com/sympy/sympy/pull/18667

**Pull request closed without merging**